src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2889,6 +2869,49 @@ struct ONS_RESOLVE : PUBLIC {
     } request;
 };
 
+/// RPC: ons/ons_info
+///
+/// Get the name mapping(s) for an Oxen Name Service entry, including metadata about the
+/// registration.  Oxen currently supports mappings for Session, Wallet and Lokinet.  Any types with
+/// a matching name hash are returned.
+///
+/// To simply resolve a record of a single type without metadata, see `ons_resolve` instead.
+///
+/// Inputs:
+///
+/// - `name_hash` -- A hashed name to look up.  See ons_resolve for details on how to properly
+///   construct this hash.  The value can be provided as either hex or base64.
+/// - `include_expired` -- Optional bool; if provided and true then expired records will be
+///   included, otherwise they will not be.
+///
+/// Outputs:
+///
+/// - `status` -- Generic RPC error code. "OK" is the success value.
+/// - `result` -- Array of found records; each one is a dict containing keys:
+///   - `type` -- the record type value (0 - session, 1 - wallet, 2 - lokinet)
+///   - `name_hash` -- the name_hash value, base64 encoded (even if hex was input).
+///   - `owner` -- the record owner, which is usually an Oxen wallet address
+///   - `backup_owner` -- a second owner; omitted if there is no backup owner.
+///   - `encrypted_value` -- the encrypted ONS record.  (See `ons_resolve`)
+///   - `expiration_height` -- if this record has an expiration, this is the height.  Omitted for
+///     non-expiring records.
+///   - `expired` -- true if this record is expired (i.e. we are past `expiration_height`), false
+///     otherwise.  Omitted for non-expiring records.
+///   - `update_height` -- the height at which this record was last modified (or created, if
+///     unmodified).
+///   - `txid` -- the transaction id that last updated (or created, if unmodified) this ONS record.
+///
+struct ONS_INFO : PUBLIC {
+    static constexpr auto names() { return NAMES("ons_info"); }
+
+    static constexpr size_t MAX_REQUEST_ENTRIES = 256;
+
+    struct request_parameters {
+        std::string name_hash;
```

Comment:
I'd prefer this to be an array, base API takes an array, helper functions abstract ontop of bulk API to do N=1. Apps can do bulk operations with minimal overhead.

If so, then `MAX_REQUEST_ENTRIES` will need to be used again in the RPC handler, or otherwise deleted from the interface.

Reply:
I've amended this some more, to bring back the ability to request multiple at once, but keeping the interface a little less awkward to use, as the RPC wallet has a call (actively used in the GUI wallet) that required either this, or adding a generic batch method.  You now can pass an array of name hashes and get:

```json
    "result": {
        "abc": [{"type": 0, ...}, {"type": 1, ...}],
        "xyz": []
    }
```
where each key corresponds to a provided input `"name_hash"` value (and the actual input, *not*, as was previously the case, the hash reconverted to base64).